src/main/java/edu/harvard/iq/dataverse/cache/CacheFactoryBean.java

```diff
@@ -0,0 +1,89 @@
+package edu.harvard.iq.dataverse.cache;
+
+import com.hazelcast.core.HazelcastInstance;
+import edu.harvard.iq.dataverse.authorization.users.User;
+import edu.harvard.iq.dataverse.util.SystemConfig;
+import jakarta.annotation.PostConstruct;
+import jakarta.ejb.EJB;
+import jakarta.ejb.Singleton;
+import jakarta.ejb.Startup;
+import jakarta.inject.Inject;
+
+import java.util.logging.Logger;
+import java.util.Map;
+
+@Singleton
+@Startup
+public class CacheFactoryBean implements java.io.Serializable {
+    private static final Logger logger = Logger.getLogger(CacheFactoryBean.class.getCanonicalName());
+    private Map<String, String> rateLimitCache;
+    @EJB
+    SystemConfig systemConfig;
+    @Inject
+    HazelcastInstance hzInstance;
+    public final static String RATE_LIMIT_CACHE = "rateLimitCache";
+
+    @PostConstruct
+    public void init() {
+        logger.info("Hazelcast member:" + hzInstance.getCluster().getLocalMember());
+        rateLimitCache = hzInstance.getMap(RATE_LIMIT_CACHE);
+        logger.info("Rate Limit Cache Size: " + rateLimitCache.size());
+    }
+
+    /**
+     * Check if user can make this call or if they are rate limited
+     * @param user
+     * @param action
+     * @return true if user is superuser or rate not limited
+     */
+    public boolean checkRate(User user, String action) {
```

Comment:
We may want to rate limit other api calls besides Command. This is why Action is passed in so it can be any api call.
The action is how it is configured, using just a string to map it to an API call